### ISSUE TITLE
Reading response body in middleware

### ISSUE THREAD

**Issue by aseipel** [tiangolo/fastapi#623]
I would like to write every response to a log before returning it.

I tried to accomplish this using middleware. However, the `call_next` method returns a `StreamingResponse`. When I read it using the same code as in `StreamingResponse.__call__`, the response obviously can't be sent to the client anymore.

Is there any way to accomplish what I'm trying to do?

**Comment by dmontagu** [comment]
Yes, it is possible; you’ll want to study this docs page: https://fastapi.tiangolo.com/tutorial/custom-request-and-route/

**Comment by aseipel** [comment]
This was super easy, thanks a lot!

**Comment by bozzlab** [comment]
> Yes, it is possible; you’ll want to study this docs page: https://fastapi.tiangolo.com/tutorial/custom-request-and-route/

Now, I can't access this page. T T

**Comment by phy25** [comment]
> > Yes, it is possible; you’ll want to study this docs page: https://fastapi.tiangolo.com/tutorial/custom-request-and-route/
> 
> Now, I can't access this page. T T

The new link should be https://fastapi.tiangolo.com/advanced/custom-request-and-route/

**Comment by dmontagu** [comment]
@phy25 Thanks for posting the new link; I've now fixed it above.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:
